workspaces/report-portal/plugins/report-portal/src/components/LaunchesPage/LaunchesPageContent/LaunchesPageContent.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect, useState } from 'react';
+
+import { ErrorPanel, Table, TableColumn } from '@backstage/core-components';
+import { useApi } from '@backstage/core-plugin-api';
+
+import Launch from '@mui/icons-material/Launch';
+import IconButton from '@mui/material/IconButton';
+import Link from '@mui/material/Link';
+import { DateTime } from 'luxon';
+
+import {
+  LaunchDetailsResponse,
+  PageType,
+  reportPortalApiRef,
+} from '../../../api';
+
+type LaunchDetails = {
+  id: number;
+  launchName: string;
+  number: number;
+  total: number;
+  passed: number;
+  failed: number;
+  skipped: number;
+  startTime: number;
+};
+
+export const LaunchesPageContent = (props: {
+  host: string;
+  project: string;
+}) => {
+  const { host, project } = props;
+  const reportPortalApi = useApi(reportPortalApiRef);
+
+  const [loading, setLoading] = useState(true);
+  const [tableData, setTableData] = useState<{
+    launches: LaunchDetails[];
+    page: PageType;
+  }>({
+    launches: [],
+    page: {
+      number: 1,
+      size: 10,
+      totalElements: 0,
+      totalPages: 1,
+    },
+  });
+
+  const [error, setError] = useState<any>();
+  useEffect(() => {
+    setLoading(true);
+    reportPortalApi
+      .getLaunchResults(project, host, {
+        'page.size': 10,
+        'page.page': 1,
+        'page.sort': 'startTime,DESC',
+      })
+      .then(res => {
+        responseHandler(res);
+      })
+      .catch(err => {
+        setLoading(false);
+        setError(err);
+      });
+  }, [host, project, reportPortalApi]);
+
+  function handlePageChange(page: number, pageSize: number) {
+    setLoading(true);
+    reportPortalApi
+      .getLaunchResults(project, host, {
+        'page.size': pageSize,
+        'page.page': page + 1,
+        'page.sort': 'startTime,DESC',
+      })
+      .then(res => {
+        responseHandler(res);
+      })
+      .catch(err => {
+        setError(err);
+        setLoading(false);
+      });
+  }
+
+  function responseHandler(res: LaunchDetailsResponse) {
+    const tempArr: LaunchDetails[] = [];
+    res.content.forEach(data => {
+      tempArr.push({
+        id: data.id,
+        launchName: data.name,
+        number: data.number,
+        total: data.statistics.executions.total ?? '-',
+        passed: data.statistics.executions.passed ?? '-',
+        failed: data.statistics.executions.failed ?? '-',
+        skipped: data.statistics.executions.skipped ?? '-',
+        startTime: data.startTime,
+      });
+    });
+    setTableData({ launches: tempArr, page: res.page });
+    setLoading(false);
+  }
+
+  const columns: TableColumn<LaunchDetails>[] = [
+    {
+      id: 0,
+      field: 'launchName',
+      title: 'Launch',
+      render: row => (
+        <Link
+          rel="noopener noreferrer"
+          target="_blank"
```

Comment:
Link from Core Components will automatically open external URLs in a new tab/window